### ISSUE TITLE
harden the build

### ISSUE THREAD

**Issue by andlabs** [andlabs/libui#6]
see https://github.com/telegramdesktop/tdesktop/issues/1112

**Comment by jvoisin** [comment]
There is a small list of things to consider [here]( https://www.owasp.org/index.php/C-Based_Toolchain_Hardening ).

**Comment by andlabs** [comment]
Thanks; I'll keep this in mind too.

**Comment by andlabs** [comment]
also use the clang static analyzer

**Comment by andlabs** [comment]
`-fsanitize=memory`

**Comment by asarubbo** [comment]
Hello.
While is fine add hardening flags, to prevent overflow to be exploited, if you talk about asan, just consider this: http://www.openwall.com/lists/oss-security/2016/02/17/9

**Comment by andlabs** [comment]
Yes, I'm aware I'm using "hardening" as a generic term here. With the cmake scripts I have set up I can make it so that asan and what not are only included in the debugging builds if I do switch them on (and I really should); thanks for the note!

**Comment by andlabs** [comment]
https://github.com/golang/go/issues/23937#issuecomment-375763928

**Comment by andlabs** [comment]
http://clang.llvm.org/docs/ControlFlowIntegrity.html

**Comment by DemiMarie** [comment]
If using clang, we *can* use `-fsanitize=undefined -fsanitize-trap-on-error` in production.  That is not only safe, but also a significant security improvement.

**Comment by andlabs** [comment]
Interesting. Do you have any more information (such as links) about those options?
